aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -204,6 +208,39 @@ private AWSAppSyncClient(AWSAppSyncClient.Builder builder) {
         mS3ObjectManager = builder.mS3ObjectManager;
     }
 
+    /**
+     * Returns the Client Subscription UUID associated with the APIKEY. Creates a new UUID if required.
+     *
+     * @param apiKey The apiKey
+     * @return The client subscription UUID.
+     */
+
+    private String getClientSubscriptionUUID(String apiKey) {
+        String clientSubscriptionUUID = null;
+        final String SHARED_PREFERENCES_FILE_NAME = "com.amazonaws.mobileconnectors.appsync";
+        try {
+            //Get Shared Preferences
+            SharedPreferences appSyncSharedPreferences = applicationContext.getSharedPreferences(SHARED_PREFERENCES_FILE_NAME, Context.MODE_PRIVATE);
+
+            //Create a SHA 256 hash of the API KEY
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            digest.reset();
+            String hash = BinaryUtils.toHex(digest.digest(apiKey.getBytes()));
+
+            clientSubscriptionUUID = appSyncSharedPreferences.getString(hash, null);
+            if (clientSubscriptionUUID == null) {
+                clientSubscriptionUUID = UUID.randomUUID().toString();
+                appSyncSharedPreferences.edit().putString(hash, clientSubscriptionUUID);
+            }
+        }
+        catch (NoSuchAlgorithmException nsae) {
```

Comment:
In this case, can we retry with a different algorithm than SHA-256 or use the default? Or should we just ignore and not use clientId?

Reply:
Should ignore as we can't safely store the APIKey.